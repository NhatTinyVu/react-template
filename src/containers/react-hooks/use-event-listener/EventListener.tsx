import React, { useState, useCallback } from 'react';
import { useEventListener } from './EventListener.hooks';
import { Coords, CoordsHandler } from './types';

const UseState: React.FC = () => {
  const [coords, setCoords]: [Coords, (coords: Coords) => void] = useState({
    x: 0,
    y: 0,
  });
  const handler: CoordsHandler = useCallback(setCoords, []);

  useEventListener({ eventName: 'mousemove', handler });

  return (
    <div>
      The mouse position is {coords.x} | {coords.y}
    </div>
  );
};

export default UseState;
