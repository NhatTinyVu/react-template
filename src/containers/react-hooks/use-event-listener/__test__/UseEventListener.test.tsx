import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import EventListener from '..';

describe('EventListener', () => {
  test('renders correctly', (): void => {
    const wrapper = shallow(<EventListener />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
