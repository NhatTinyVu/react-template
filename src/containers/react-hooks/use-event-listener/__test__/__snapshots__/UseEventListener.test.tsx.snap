// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListener renders correctly 1`] = `
<div>
  The mouse position is 
  0
   | 
  0
</div>
`;
